src/server/service/match.ts

```diff
@@ -174,6 +176,31 @@ export class Match {
         });
       }
 
+      if (!player.isBot) {
+        // TODO: Implement player stats functionality
+        //const playerStats = getPlayerStats(player.id);
+
+        // Check achievements
+        const achievementPoints = Object.entries(MATCH_ACHIEVEMENTS)
+          .map(([achievementId, achievement]) => {
+            const pointsEarned = achievement.calculate({
+              player,
+              messages: this._messages,
+            });
+            return { achievementId, pointsEarned };
```

Comment:
you can call these `id` and `points` and later directly `.push(achievement)` in the reducer, without reconstructing the props